internal/regexpchecker/regexpchecker.go

```diff
@@ -647,7 +674,7 @@ func (v *regExpValidator) scanCharacterEscape(atomEscape bool) string {
 		}
 		if v.anyUnicodeModeOrNonAnnexB {
 			v.error(diagnostics.X_c_must_be_followed_by_an_ASCII_letter, v.pos-2, 2)
-		} else if atomEscape {
+		} else {
```

Comment:
This is weird, but in Annex B, `\c` are really two characters regardless of whether it is an atom escape. I believe it is due to the [ClassAtomNoDash](https://tc39.es/ecma262/#prod-annexB-ClassAtomNoDash) production.
Also some changes from https://github.com/microsoft/TypeScript/pull/60249; didn't file a separate PR due to insignificance.